### PR TITLE
 Fix Windows environment issues

### PR DESCRIPTION
**Pull request by yuslepukhin**
Enable directIO on WritableFileImpl::Append
     with offset being current length of the file.
     Enable UniqueID tests on Windows, disable others but
     leeting them to compile. Unique tests are valuable to
     detect failures on different filesystems and upcoming
     ReFS.
     Clear output in WinEnv Getchildren.This is different from
     previous strategy, do not touch output on failure.
     Make sure DBTest.OpenWhenOpen works with windows error message

### CODE REVIEW

**Comment by yuslepukhin** [comment]
single failure unrelated to the change. The other build got terminated due to exceeding time limit. When can I expect to see this in master?

**Comment by IslamAbdelRahman** [comment]
@yuslepukhin will take a look soon

File: port/win/io_win.cc [hunk 1]
```diff
@@ -847,29 +873,64 @@ WinWritableImpl::WinWritableImpl(WinFileData* file_data, size_t alignment)
...
+  uint64_t written = 0;
 
-  DWORD bytesWritten = 0;
-  if (!WriteFile(file_data_->GetFileHandle(), data.data(),
-    static_cast<DWORD>(data.size()), &bytesWritten, NULL)) {
-    auto lastError = GetLastError();
-    s = IOErrorFromWindowsError(
-      "Failed to WriteFile: " + file_data_->GetName(),
-      lastError);
+  if (file_data_->UseDirectIO()) {
```
Comment (IslamAbdelRahman) [inline port/win/io_win.cc]: What changed the `assert(!file_data_->UseDirectIO());` ?
Where are we using AppendImpl with DirectIO enabled

Comment (yuslepukhin) [inline port/win/io_win.cc]: This is being used within the env_test and this is what actually generates test failure. I have decided to change it inline with 1) POSIX impl and 2) New documentation for Env::Append() which states that append must write to the end of the file and thus no offset is necessary

**Pull request closed without merging**